src/notes/notes.service.spec.ts

```diff
@@ -67,9 +79,803 @@ describe('NotesService', () => {
+      const newNote = await service.createNote(content);
+      const revisions = await newNote.revisions;
+      jest.spyOn(revisionRepo, 'findOne').mockResolvedValueOnce(revisions[0]);
+      service.getLatestRevision(newNote).then((result) => {
+        expect(result).toEqual(revisions[0]);
+      });
+    });
+  });
+
+  describe('getNoteByIdOrAlias', () => {
+    it('works', async () => {
+      const user = User.create('hardcoded', 'Testy') as User;
+      const note = Note.create(user);
+      jest.spyOn(noteRepo, 'findOne').mockResolvedValueOnce(note);
+      const foundNote = await service.getNoteByIdOrAlias('noteThatExists');
+      expect(foundNote).toEqual(note);
+    });
+    it('fails: no note found', async () => {
+      jest.spyOn(noteRepo, 'findOne').mockResolvedValueOnce(undefined);
+      try {
+        await service.getNoteByIdOrAlias('noteThatDoesNoteExist');
+      } catch (e) {
+        expect(e).toBeInstanceOf(NotInDBError);
+      }
+    });
+  });
+
+  describe('deleteNoteByIdOrAlias', () => {
+    it('works', async () => {
+      const user = User.create('hardcoded', 'Testy') as User;
+      const note = Note.create(user);
+      jest.spyOn(noteRepo, 'findOne').mockResolvedValueOnce(note);
+      jest
+        .spyOn(noteRepo, 'remove')
+        .mockImplementationOnce(async (entry, _) => {
+          expect(entry).toEqual(note);
+          return entry;
+        });
+      await service.deleteNoteByIdOrAlias('noteThatExists');
+    });
+  });
+
+  describe('updateNoteByIdOrAlias', () => {
+    it('works', async () => {
+      const user = User.create('hardcoded', 'Testy') as User;
+      const note = Note.create(user);
+      const revisionLength = (await note.revisions).length;
+      jest.spyOn(noteRepo, 'findOne').mockResolvedValueOnce(note);
+      jest
+        .spyOn(noteRepo, 'save')
+        .mockImplementationOnce(async (entry: Note) => {
+          return entry;
+        });
+      const updatedNote = await service.updateNoteByIdOrAlias(
+        'noteThatExists',
+        'newContent',
+      );
+      expect(await updatedNote.revisions).toHaveLength(revisionLength + 1);
+    });
+  });
+
+  describe('updateNotePermissions', () => {
+    const userPermissionUpdate = new NoteUserPermissionUpdateDto();
+    userPermissionUpdate.username = 'hardcoded';
+    userPermissionUpdate.canEdit = true;
+    const groupPermissionUpate = new NoteGroupPermissionUpdateDto();
+    groupPermissionUpate.groupname = 'testGroup';
+    groupPermissionUpate.canEdit = false;
+    const user = User.create(userPermissionUpdate.username, 'Testy') as User;
+    const group = Group.create(
+      groupPermissionUpate.groupname,
+      groupPermissionUpate.groupname,
+    );
+    const note = Note.create(user);
+    describe('works', () => {
+      it('with empty GroupPermissions and with empty UserPermissions', async () => {
+        jest.spyOn(noteRepo, 'findOne').mockResolvedValueOnce(note);
```

Comment:
Maybe move the mocking into a `beforeEach`?

Reply:
This is not possible as the resolved value is not the same in all tests…